wasm/lib/src/wasmer_locator.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:io';
+
+/// Attempts to locate the Wasmer dynamic library.
+///
+/// Throws [WasmLocatorError] if the dynamic library could not be found.
+DynamicLibrary loadWasmerDynamicLibrary() {
+  String? locate(String libName) {
```

Comment:
You can eliminate a lot of the duplicated code below if you have this function return a non-null String, and throw the WasmLocatorError in here.

Reply:
I agree, but notice how iOS and Android can't fail to be located and how each error message refers to the platform it belongs to.

Handling those that can fail in a separate if statement would remove the clarity that we get here from being explicit. Furthermore, this allows us to easily see that we are being exhaustive with out platform checks and are essentially done here. 

So I think, even if there's some duplication here, for the purposes of clearly communicating the intent of this code to the reader, it should be tolerated.

Reply:
You can still have this chain of if statements, so iOS and Android loading would look the same, and you still get the code clarity. I'm just saying throw the exception inside `locate`. The only thing you loose is telling the user what platform they're on, which they presumably already know.

Reply:
I've applied your suggested changes.